custom_verb_test.go

```diff
@@ -0,0 +1,61 @@
+package restful
+
+import "testing"
+
+func TestHasCustomVerb(t *testing.T) {
+	testCase := []struct{
+		path string
+		has bool
+	}{
+		{"/{userId}:init", true},
+		{"/{userId:init}", false},
+		{"/users/{id:init}:init", true},
+		{"/users/{id}", false},
+	}
+
+	for _, v := range  testCase {
+		rs := hasCustomVerb(v.path)
+		if rs != v.has {
+			t.Errorf("path: %v should has no custom verb", v.path)
+		}
+	}
+}
+
+func TestRemoveCustomVerb(t *testing.T) {
+	testCase := []struct{
+		path string
+		expectedPath string
+	}{
+		{"/{userId}:init", "/{userId}"},
+		{"/{userId:init}", "/{userId:init}"},
+		{"/users/{id:init}:init", "/users/{id:init}"},
+		{"/users/{id}", "/users/{id}"},
```

Comment:
this will fail:   "/init/users/{id:init}:init" , "/init/users/{id:init}"

Reply:
i added it to case, and run success,  is something wrong?
```
	testCase := []struct{
		path string
		expectedPath string
	}{
		{"/{userId}:init", "/{userId}"},
		{"/{userId:init}", "/{userId:init}"},
		{"/users/{id:init}:init", "/users/{id:init}"},
		{"/users/{id}", "/users/{id}"},
		{"/init/users/{id:init}:init" , "/init/users/{id:init}"},
	}
```

Reply:
tbh, i did not execute this test. i should have said "will this fail?".  I came up with this test because I saw "ReplaceAllString"  in removeCustomVerb.   And this case "/:init/users:init/" is probably not a valid path, right?

Reply:
emmm, i think it likes "/users/{id:init}:init", so i think the case is not necessary.
and replace regex is ":customVerb$", it will just replace last word.

Reply:
ok, good!